### ISSUE TITLE
Publish Repository button does not push tags

### ISSUE THREAD

**Issue by itagagaki** [desktop/desktop#17410]
### The problem

I created a local repository, created tags in GitHub Desktop, and created a remote repository on GitHub.com using the 'Publish repository' button, but the tags are not reflected.

### Release version

3.3.1 (x64)

### Operating system

Windows 10 Pro 22H2

**Comment by steveward** [comment]
@itagagaki thanks for the report. I tested this out and was unable to reproduce the issue. The tags I created in GitHub Desktop were pushed up when publishing the repository. Can you confirm the tags were created in GitHub Desktop? GitHub Desktop only supports pushing tags that have been created in the app itself.

**Comment by itagagaki** [comment]
@steveward Thank you for responding so quickly.

I tried the same procedure to reproduce it, but could not reproduce it. The tag was also pushed. Strange...
Of course, I created the tags in the history section of the GitHub Desktop. But as you can see in this image, the tags were not pushed yesterday.

![image](https://github.com/desktop/desktop/assets/71218184/8d665a0c-ad5f-4620-b81b-656c6f95c3dd)

If we can't reproduce this, we can't find the cause. It can't be helped. You can close the issue as "not reproduced".

However, if there is a way to reflect the tags in the local repository (as seen in the GitHub desktop) to the current remote repository, I would be grateful if you could let me know.

Thank you.

**Comment by steveward** [comment]
Thanks for the update.

As a workaround, if your workflow requires that you push tags created by external tools (CLI, build scripts etc) you could try setting the global (or per-repo) config setting [push.followTags](https://git-scm.com/docs/git-config#Documentation/git-config.txt-pushfollowTags) which will push all tags on every push (regardless of the push being done through the CLI or through GitHub Desktop). I hope that helps!

**Comment by itagagaki** [comment]
@steveward No, I do not create tags with any external tool or CLI. All tags were created in the History section of the GitHub Desktop using the right-click context menu.

Is there any way to get the remote repository to be exactly as it appears in GitHub Desktop, or do I have to recreate the tags on github.com? Is that okay? In other words, can I keep it consistent with the local repository?

**Comment by itagagaki** [comment]
I just tried right-clicking on a commit "0.8.1" in the History section of the GitHub Desktop and noticed that there's a "Create Tag..." button and the "Delete tag 0.8.1" button is disabled.
This is really weird because these tags were actually created using the "Create Tag..." button in GitHub Desktop.

What if I use "Create Tag..." again in this state? Do you think it is safe?

**Comment by steveward** [comment]
A tag that has been pushed to GitHub cannot be deleted, see [this explanation](https://github.com/desktop/desktop/issues/9938#issuecomment-638779503).

You can create multiple tags on the same commit without issue.

**Comment by itagagaki** [comment]
That topic is about tags that have BEEN PUSHED to GitHub, right?
My problem is that the tags I have created did NOT get pushed.
Please see the screenshot above. GitHub (right side) shows "0 tags".

I currently have a commit "0.8.1" in my local repository with a tag "0.8.1". (As I have said many times before, these tags were created using GitHub Desktop). Can I create a tag with the same name again in GitHub Desktop for this commit? That is not the same as multiple tags, is it?

**Comment by steveward** [comment]
Can you try opening the repository in the command line and run `git push origin --tags` to push those tags up? I'm not sure why GitHub Desktop is failing to push those tags if they were created in the app itself.

**Comment by itagagaki** [comment]
That made it work. Thank you very much.

So my remote repository is now in the same state as if I had used the CLI on the local repository to create and push the tags. I finally got the result I wanted, but the cause of the GitHub Desktop failure is still a mystery, so it is a bit spooky.

This issue is probably a rare case, as I cannot reproduce it. I don't think there's any way to investigate it, so I hope it doesn't happen again.

Thanks for your help so far!
